app/src/main/java/io/appium/uiautomator2/handler/FirstVisibleView.java

```diff
@@ -0,0 +1,94 @@
+package io.appium.uiautomator2.handler;
+
+import android.support.test.uiautomator.StaleObjectException;
+import android.support.test.uiautomator.UiObject;
+import android.support.test.uiautomator.UiObject2;
+import android.support.test.uiautomator.UiObjectNotFoundException;
+import android.support.test.uiautomator.UiSelector;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.UUID;
+
+import io.appium.uiautomator2.common.exceptions.UiAutomator2Exception;
+import io.appium.uiautomator2.core.AccessibilityNodeInfoGetter;
+import io.appium.uiautomator2.handler.request.SafeRequestHandler;
+import io.appium.uiautomator2.http.AppiumResponse;
+import io.appium.uiautomator2.http.IHttpRequest;
+import io.appium.uiautomator2.model.AndroidElement;
+import io.appium.uiautomator2.model.KnownElements;
+import io.appium.uiautomator2.server.WDStatus;
+import io.appium.uiautomator2.utils.Logger;
+
+import static io.appium.uiautomator2.utils.Device.getAndroidElement;
+
+/**
+ * This method return first visible element inside provided element
+ */
+public class FirstVisibleView extends SafeRequestHandler {
+
+    public FirstVisibleView(String mappedUri) {
+        super(mappedUri);
+    }
+
+    @Override
+    public AppiumResponse safeHandle(IHttpRequest request) {
+        Logger.info("Get first visible element inside provided element");
+        String elementId = getElementId(request);
+
+        AndroidElement element = KnownElements.getElementFromCache(elementId);
+        if (element == null) {
+            return new AppiumResponse(getSessionId(request), WDStatus.NO_SUCH_ELEMENT);
+        }
+        try {
+            KnownElements ke = new KnownElements();
+            Object firstObject = null;
+            if (element.getUiObject() instanceof UiObject) {
+                Logger.debug("Container for first visible is a uiobject");
+                UiObject childObject = ((UiObject) element.getUiObject()).getChild(new UiSelector().index(0));
```

Comment:
this might fail if the element has zero children

Reply:
wouldn't it then be caught by the handling on line 83, and an element not found error appropriately returned?